com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/util/CloudModelBuilderUtil.java

```diff
@@ -74,17 +75,31 @@ public static String buildImplicitDeployTargetName(String org, String space) {
         return String.format("%s %s", org, space);
     }
 
-    public static boolean isService(Resource resource) {
-        return resource.getType() != null; // Typed resource = service;
+    public static boolean isService(Resource resource, PropertiesAccessor propertiesAccessor) {
+        if (resource.getType() == null) {
+            return false;
+        }
+        return !ResourceType.EXISTING_SERVICE_KEY.toString().equals(getResourceType(resource, propertiesAccessor));
+    }
+
+    public static boolean isServiceKey(Resource resource, PropertiesAccessor propertiesAccessor) {
+        if (resource.getType() == null) {
+            return false;
+        }
+        return ResourceType.EXISTING_SERVICE_KEY.toString().equals(getResourceType(resource, propertiesAccessor));
     }
 
     public static <R> R parseParameters(List<Map<String, Object>> parametersList, ParametersParser<R> parser) {
         return parser.parse(parametersList);
     }
 
-    public static ServiceType getServiceType(Map<String, Object> properties) {
-        String type = (String) properties.getOrDefault(SupportedParameters.TYPE, ServiceType.MANAGED.toString());
-        return ServiceType.get(type);
+    public static ResourceType getServiceType(Map<String, Object> properties) {

```

Comment:
Could we rename this to getResourceType?

Reply:
Yes, it will be changed.